src/auth/client/mod.rs

```diff
@@ -78,48 +80,159 @@ pub enum LoginResponse {
 		redirect_address: (LuString33, u16),
 		/// Whether the account is in free trial mode.
 		is_ftp: bool,
+		/// Whether the account is connecting as a paid account for the first time.
+		just_upgraded_from_ftp: bool,
+		language: Language,
+		country_code: LuString3,
+		auth_logs: Vec<LogEntry>,
```

Comment:
I would add docstrings to these but am not certain what they actually contain. I'm guessing language is about the server language (relevant for player interaction), country code is probably where the server is but I'm not sure how that's relevant, and what the auth logs actually mean I have no idea about.

Reply:
It's not required, but if you want, you could take existing descriptions from https://docs.google.com/document/d/e/2PACX-1vQ8qUS7z37juAY_5vkfvxlBHpkXQSRycD1fyBLyeG9IdflqM1BBqFe2zGNju6ZJzK5nIHHejZaAxt76/pub#h.n00rdwx4rhf0.